File: clean-architecture/python/order/OrderPlaced.py

from typing import List, Union
from AbstractEvent import AbstractEvent
import json
from datetime import datetime


class OrderPlaced(AbstractEvent):
    id : int
    productId : int
    qty : int
    productName : str

    def __init__(self):
    	super().__init__()
    	self.id = None
    	self.productId = None
    	self.qty = None
    	self.productName = None
		

File: clean-architecture/go/shop/order/order/OrderPlaced.go
package order

import (
	"time"
)

type OrderPlaced struct {
	EventType   string `json:"eventType" type:"string"`
	TimeStamp   string `json:"timeStamp" type:"string"`
	Id          int    `json:"id" type:"int"`
	ProductId   int    `json:"productId" type:"int"`
	Qty         int    `json:"qty" type:"int"`
	ProductName string `json:"productName" type:"string"`
}

func NewOrderPlaced() *OrderPlaced {
	event := &OrderPlaced{EventType: "OrderPlaced", TimeStamp: time.Now().String()}

	return event
}
